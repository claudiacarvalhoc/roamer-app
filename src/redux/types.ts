export const ADD_PROJECT = 'ADD_PROJECT';
export const ADD_LANGUAGES = 'ADD_LANGUAGES';
export const REMOVE_TRANSLATION = 'REMOVE_TRANSLATION';
export const EXPAND_PROJECTS = 'EXPAND_PROJECTS';
export const COLLAPSE_PROJECTS = 'COLLAPSE_PROJECTS';