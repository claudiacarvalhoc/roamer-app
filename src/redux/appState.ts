import { DefaultRootState } from "react-redux";

export interface AppState extends DefaultRootState {
    expand: boolean;
    projects: ProjectProperties[];
    menuTexts: MenuTexts,
    projectTexts: ProjectTextsProperties,
}

export interface MenuTexts {
    projectText: string;
    expandText: string;
    collapseText: string;
}

export interface ProjectTextsProperties {
    doneText: string;
    baseWordsText: string;
    teamText: string;
    keysText: string;
    qaIssuesText: string;
}

export interface IconType {
    Icon: 'up' | 'down' | 'check' | 'user' | 'rate' | 'camera' | 'book';
}

export interface TranslationSectionsProperties {
    country: string;
    language: string;
    done: number;
    wordsToDo: number;
    unverified: number;
}

export interface ProjectProperties {
    id: number;
    name: string;
    done: number;
    baseWords: number;
    team: number;
    keys: number;
    issues: number;
    icons: string[];
    badges: string[];
    translationSections: TranslationSectionsProperties[];
}
