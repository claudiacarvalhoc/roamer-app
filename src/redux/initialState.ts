import { AppState, MenuTextsState, ProjectState, ProjectTextsState } from "./appState";

export const menuInitialState: MenuTextsState = {
    projectText: 'New project ⇧⌘P',
    expandText: 'Expand all',
    collapseText: 'Collapse all',
};

export const projectsInitialState: ProjectTextsState = {
    doneText: 'DONE',
    baseWordsText: 'BASE WORDS',
    teamText: 'TEAM',
    keysText: 'KEYS',
    qaIssuesText: 'QA ISSUES',
    wordsToDoText: 'WORDS TO DO',
    unverifiedText: 'UNVERIFIED',
    addLanguageText: 'Add language',
};

export const projectInitialState: ProjectState = {
    id: 1,
    name: 'Roamer App (Android SDK test)',
    done: 30,
    baseWords: 160246,
    team: 26,
    keys: 1834,
    issues: 47514,
    icons: ['up', 'down', 'check', 'user', 'rate', 'camera', 'book'],
    badges: [
        {
            name: 'Roamer',
            type: 'orange',
        },
        {
            name: 'iOS',
            type: 'green',
        }
    ],
    translationSections: [{
        id: 1,
        country: 'uk',
        language: 'English',
        done: 20,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 2,
        country: 'uk',
        language: 'English',
        done: 20,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 3,
        country: 'uk',
        language: 'English',
        done: 20,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 4,
        country: 'uk',
        language: 'English',
        done: 20,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 5,
        country: 'ru',
        language: 'Russian',
        done: 45,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 6,
        country: 'ru',
        language: 'Russian',
        done: 45,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 7,
        country: 'ru',
        language: 'Russian',
        done: 45,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 8,
        country: 'ru',
        language: 'Russian',
        done: 45,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 9,
        country: 'it',
        language: 'Italian',
        done: 83,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 10,
        country: 'it',
        language: 'Italian',
        done: 83,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 11,
        country: 'it',
        language: 'Italian',
        done: 83,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 12,
        country: 'it',
        language: 'Italian',
        done: 83,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 13,
        country: 'mk',
        language: 'Macedonian',
        done: 67,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 14,
        country: 'mk',
        language: 'Macedonian',
        done: 67,
        wordsToDo: 16246,
        unverified: 50000,
    }, {
        id: 15,
        country: 'mk',
        language: 'Macedonian',
        done: 67,
        wordsToDo: 16246,
        unverified: 50000,
    }]
};

export const appInitialState: AppState = {
    expand: true,
    projects: [
        projectInitialState
    ],
    MenuTextsState: menuInitialState,
    projectTexts: projectsInitialState,
};

export const newProjectInitialState: ProjectState = {
    id: 0,
    name: 'Roamer App (Android SDK test)',
    done: 0,
    baseWords: 160246,
    team: 26,
    keys: 1834,
    issues: 47514,
    icons: ['up', 'down', 'check', 'user', 'rate', 'camera', 'book'],
    badges: [
        {
            name: 'Roamer',
            type: 'orange',
        },
        {
            name: 'iOS',
            type: 'green',
        }
    ],
    translationSections: [],
};