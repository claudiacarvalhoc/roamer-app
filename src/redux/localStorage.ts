import { AppState } from './appState';
const KEY = 'state';

export const loadState = (): AppState => {
    try {
        const serializedState = localStorage.getItem(KEY);
        return serializedState !== null ? JSON.parse(serializedState) : undefined;
    } catch (err) {
        console.error(`Error loading state from local storage`, err);
        return undefined;
    }
};

export const saveState = (state: AppState) => {
    try {
        const serializedState = JSON.stringify(state);
        localStorage.setItem(KEY, serializedState);
    } catch (err) {
        console.error(`Error saving state on local storage`, err);
    }
};