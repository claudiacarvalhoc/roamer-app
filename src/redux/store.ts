import { AnyAction } from 'redux';
import { AppState } from './appState';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import { createLogger } from 'redux-logger';
import { rootReducer } from './reducers';
import thunk from 'redux-thunk';
import { loadState, saveState } from './localStorage';

export type AppThunk<R> = ThunkAction<R, AppState, unknown, AnyAction>;
export type AppDispatch = ThunkDispatch<AppState, unknown, AnyAction>;


export const configureStore = (initialState: AppState) => {
    const logger = createLogger();
    const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const middleware = [thunk, logger];

    const preloadedState = loadState() ?? initialState;
    const store = createStore(
        rootReducer,
        {
            app: preloadedState,
        },
        composeEnhancers(applyMiddleware(...middleware)),
    );

    store.subscribe(() => {
        saveState(store.getState().app);
    });

    return store;
};

export default configureStore;
