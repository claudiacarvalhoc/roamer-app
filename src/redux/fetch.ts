import { AppDispatch } from './store';
import { addProjectAction } from './app/actions';
import { newProjectInitialState } from './initialState';

export const createProject = (): ((dispatch: AppDispatch) => Promise<void>) => {
    return dispatch => {
        dispatch(addProjectAction(newProjectInitialState));
        return Promise.resolve();
    };
};