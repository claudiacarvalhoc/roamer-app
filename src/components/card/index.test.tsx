import React from 'react';
import Card, { CardProps } from './index';
import configureMockStore from 'redux-mock-store';
import Enzyme, { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { appInitialState } from '../../redux/initialState';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

const getShallowWrapper = (props) => {
    const store = configureMockStore()({
        app: appInitialState
    });
    return mount(
        <Provider store={store}>
            <Card {...props} />
        </Provider>
    );
};

Enzyme.configure({ adapter: new Adapter() });

const props: CardProps = {
    className: 'test',
    translation: {
        id: 1,
        country: 'uk',
        language: 'English',
        done: 10,
        wordsToDo: 50000,
        unverified: 30000
    },
    texts: {
        doneText: 'DONE',
        baseWordsText: 'BASE WORDS',
        teamText: 'TEAM',
        keysText: 'KEYS',
        qaIssuesText: 'QA ISSUES',
        wordsToDoText: 'WORDS TO DO',
        unverifiedText: 'UNVERIFIED',
        addLanguageText: 'Add language',
    }
};

describe('<Card />', () => {
    it('render correctly', () => {
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
});
