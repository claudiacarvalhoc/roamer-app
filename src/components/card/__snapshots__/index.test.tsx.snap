// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Card)
    className="test"
    texts={
      Object {
        "addButtonText": "Add",
        "addLanguageText": "Add language",
        "baseWordsText": "BASE WORDS",
        "closeButtonText": "Close",
        "doneText": "DONE",
        "keysText": "KEYS",
        "modalTitleText": "Add Languages",
        "qaIssuesText": "QA ISSUES",
        "teamText": "TEAM",
        "unverifiedText": "UNVERIFIED",
        "wordsToDoText": "WORDS TO DO",
      }
    }
    translation={
      Object {
        "country": "uk",
        "done": 10,
        "id": 1,
        "language": "English",
        "unverified": 30000,
        "wordsToDo": 50000,
      }
    }
  >
    <Card
      className="test"
      dispatch={[Function]}
      texts={
        Object {
          "addButtonText": "Add",
          "addLanguageText": "Add language",
          "baseWordsText": "BASE WORDS",
          "closeButtonText": "Close",
          "doneText": "DONE",
          "keysText": "KEYS",
          "languagesText": Array [
            Object {
              "label": "Russian",
              "value": "ru",
            },
            Object {
              "label": "German",
              "value": "de",
            },
            Object {
              "label": "English",
              "value": "uk",
            },
            Object {
              "label": "French",
              "value": "fr",
            },
            Object {
              "label": "Italian",
              "value": "it",
            },
            Object {
              "label": "Spanish",
              "value": "es",
            },
            Object {
              "label": "Ukrainian",
              "value": "ua",
            },
            Object {
              "label": "Poland",
              "value": "pl",
            },
            Object {
              "label": "Polish",
              "value": "ro",
            },
            Object {
              "label": "Dutch",
              "value": "nl",
            },
            Object {
              "label": "Czech",
              "value": "cz",
            },
            Object {
              "label": "Greek",
              "value": "gr",
            },
            Object {
              "label": "Portuguese",
              "value": "pt",
            },
            Object {
              "label": "Swedish",
              "value": "se",
            },
            Object {
              "label": "Hungarian",
              "value": "bu",
            },
            Object {
              "label": "Belarusian",
              "value": "by",
            },
            Object {
              "label": "Serbian",
              "value": "rs",
            },
            Object {
              "label": "Bulgarian",
              "value": "bg",
            },
            Object {
              "label": "Danish",
              "value": "dk",
            },
            Object {
              "label": "Finnish",
              "value": "fi",
            },
            Object {
              "label": "Macedonian",
              "value": "mk",
            },
          ],
          "modalTitleText": "Add Languages",
          "qaIssuesText": "QA ISSUES",
          "teamText": "TEAM",
          "unverifiedText": "UNVERIFIED",
          "wordsToDoText": "WORDS TO DO",
        }
      }
      translation={
        Object {
          "country": "uk",
          "done": 10,
          "id": 1,
          "language": "English",
          "unverified": 30000,
          "wordsToDo": 50000,
        }
      }
    >
      <div
        className="test container"
      >
        <Language
          flag="uk"
          name="English"
        >
          <div
            className="container"
          >
            <UnitedKingdomFlag
              className="flag"
              key="vector-uk"
            >
              <svg
                className="flag"
                fill="none"
                height="10"
                viewBox="0 0 14 10"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="white"
                  height="10"
                  rx="1"
                  width="14"
                />
                <mask
                  height="10"
                  id="mask0"
                  mask-type="alpha"
                  maskUnits="userSpaceOnUse"
                  width="14"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="white"
                    height="10"
                    rx="1"
                    width="14"
                  />
                </mask>
                <g
                  mask="url(#mask0)"
                >
                  <rect
                    fill="#0A17A7"
                    height="10"
                    width="14"
                  />
                  <path
                    clipRule="evenodd"
                    d="M-0.640625 -0.958496L5.33333 3.07099V-0.666992H8.66667V3.07161L14.6413 -0.958361L15.3869 0.147022L10.6635 3.33301H14V6.66634H10.6637L15.3874 9.85257L14.6418 10.958L8.66667 6.92764V10.6663H5.33333V6.92826L-0.641128 10.9581L-1.38672 9.8527L3.33726 6.66634H0V3.33301H3.3374L-1.38622 0.146887L-0.640625 -0.958496Z"
                    fill="white"
                    fillRule="evenodd"
                  />
                  <path
                    d="M9.33333 3.1661L15.666 -1"
                    stroke="#DB1F35"
                    strokeLinecap="round"
                    strokeWidth="0.666667"
                  />
                  <path
                    d="M10.0065 6.84871L15.6834 10.6751"
                    stroke="#DB1F35"
                    strokeLinecap="round"
                    strokeWidth="0.666667"
                  />
                  <path
                    d="M4.6445 6.80232L-1.91927 11.1551"
                    stroke="#DB1F35"
                    strokeLinecap="round"
                    strokeWidth="0.666667"
                  />
                  <path
                    clipRule="evenodd"
                    d="M0 6H6V10H8V6H14V4H8V0H6V4H0V6Z"
                    fill="#E6273E"
                    fillRule="evenodd"
                  />
                </g>
              </svg>
            </UnitedKingdomFlag>
            <p
              className="language"
            >
              English
            </p>
          </div>
        </Language>
        <div
          onClick={[Function]}
        >
          <Icons
            className="trash"
            vectors={
              Array [
                "trash",
              ]
            }
          >
            <div
              className="trash container"
            >
              <TrashVector
                className="icon"
                key="vector-trash"
              >
                <svg
                  className="icon"
                  height="15"
                  id="Capa_1"
                  version="1.1"
                  viewBox="0 0 488.936 488.936"
                  x="0px"
                  xmlns="http://www.w3.org/2000/svg"
                  y="0px"
                >
                  <g>
                    <g>
                      <path
                        d="M381.16,111.948H107.376c-6.468,0-12.667,2.819-17.171,7.457c-4.504,4.649-6.934,11.014-6.738,17.477l9.323,307.69 c0.39,12.92,10.972,23.312,23.903,23.312h20.136v-21.012c0-24.121,19.368-44.049,43.488-44.049h127.896 c24.131,0,43.893,19.928,43.893,44.049v21.012h19.73c12.933,0,23.52-10.346,23.913-23.268l9.314-307.7 c0.195-6.462-2.234-12.863-6.738-17.513C393.821,114.767,387.634,111.948,381.16,111.948z"
                      />
                      <path
                        d="M309.166,435.355H181.271c-6.163,0-11.915,4.383-11.915,11.516v30.969c0,6.672,5.342,11.096,11.915,11.096h127.895 c6.323,0,11.366-4.773,11.366-11.096v-30.969C320.532,440.561,315.489,435.355,309.166,435.355z"
                      />
                      <path
                        d="M427.696,27.106C427.696,12.138,415.563,0,400.591,0H88.344C73.372,0,61.239,12.138,61.239,27.106v30.946 c0,14.973,12.133,27.106,27.105,27.106H400.59c14.973,0,27.105-12.133,27.105-27.106L427.696,27.106L427.696,27.106z"
                      />
                    </g>
                  </g>
                </svg>
              </TrashVector>
            </div>
          </Icons>
        </div>
        <ProgressBar
          className="progressBar"
          percentage={10}
        >
          <div
            className="progressBar"
          >
            <div
              style={
                Object {
                  "border": "2px solid #CFCFCF",
                  "height": 0,
                  "width": "100%",
                }
              }
            >
              <div
                style={
                  Object {
                    "border": "2px solid #DB1F35",
                    "marginLeft": "-2px",
                    "marginTop": "-2px",
                    "width": "10%",
                  }
                }
              />
            </div>
          </div>
        </ProgressBar>
        <div
          className="labels"
        >
          <Label
            highlighted={false}
            name="DONE"
            value="10%"
          >
            <div
              className="label"
            >
              <p
                className="name"
              >
                DONE
              </p>
              <p
                className="value"
              >
                10%
              </p>
            </div>
          </Label>
          <Label
            highlighted={true}
            name="WORDS TO DO"
            value="50000"
          >
            <div
              className="label"
            >
              <p
                className="name"
              >
                WORDS TO DO
              </p>
              <p
                className="value highlighted"
              >
                50000
              </p>
            </div>
          </Label>
          <Label
            highlighted={true}
            name="UNVERIFIED"
            value="30000"
          >
            <div
              className="label"
            >
              <p
                className="name"
              >
                UNVERIFIED
              </p>
              <p
                className="value highlighted"
              >
                30000
              </p>
            </div>
          </Label>
        </div>
      </div>
    </Card>
  </Connect(Card)>
</Provider>
`;
